sdk-core/transaction.md

```diff
@@ -44,6 +44,17 @@ dto DraftTransaction:
 ```
 class TransactionComputer:
     compute_transaction_fee(transaction: Transaction, network_config: INetworkConfig): Amount;
+
     compute_bytes_for_signing(transaction: Transaction): bytes;
+
     compute_transaction_hash(transaction: Transaction): bytes;
+
+    // checks if the second least significant bit is set; returns True if it's set, False otherwise
+    is_guarded_transaction(transaction: Transaction): bool;
+
+    // sets guardian address, guardian signature, transaction.version = 2, sets transaction.options second least significant bit
+    apply_guardian(
```

Comment:
`transaction` should also be fed as input. If the received object is mutated, then maybe do not return the object (if already fed as input)?

Also, normally, `transaction.guardian` should have been set in advance, prior serialization for signing. Thus, this function would be, in practice, I think, a simple:

```
transaction.guardian_signature = ...
```

Same for `options`, `version` (should be set prior signing).

Or, perhaps, `apply_guardian` should set `version` `options`, but not `guardian_signature`, instead?

Reply:
@ccorcoveanu, all right to hold `apply_guardian` here, in the `TransactionComputer`, which acts as a holder of behavior related to transactions (while the `Transaction` class itself is a mere DTO)?

I'd vote that "yes". The name "computer" is sufficiently generic.

We use this "pattern" here in the specs, and in PY and JS already: dummy DTOs, with adjacent `*Computer`.

Reply:
Indeed. The `guardian_signature` should not be set by `apply_guardian()`.

Reply:
@andreibancioiu yup, I am ok with this approach